added new category 'troubleshooting' and moved 4 faqs to it (#724)
NixOS/nix.dev: source/recipes/troubleshooting.md

# Troubleshooting

## What to do if a binary cache is down or unreachable?

> Pass `--option substitute false` to Nix commands.

Reviewer: ```suggestion
Pass [`--option substitute false`](https://nixos.org/manual/nix/stable/command-ref/conf-file#conf-substitute) to Nix commands.
```

Pro tip: Whenever you have some command or term that is not defined in context, link to reference documentation. How are you going to find that? Well, how are readers going to find it without a link? Ask reviewers if you need help :)

> ### How do I fix: error: querying path in database: database disk image is malformed

Reviewer: ```suggestion
### How to fix: `error: querying path in database: database disk image is malformed`
```

> Try:

Reviewer: ```suggestion
This is a [known issue](https://github.com/NixOS/nix/issues/1353).
Try:
```

```shell-session
$ sqlite3 /nix/var/nix/db/db.sqlite "pragma integrity_check"
```

> Which will print the errors in the database. If the errors are due to missing
> references, the following may work:

Reviewer: ```suggestion
Which will print the errors in the [database](https://nixos.org/manual/nix/stable/glossary#gloss-nix-database).
If the errors are due to missing references, the following may work:
```

Use one line per sentence, easier to review and make suggestions.

```shell-session
$ mv /nix/var/nix/db/db.sqlite /nix/var/nix/db/db.sqlite-bkp
$ sqlite3 /nix/var/nix/db/db.sqlite-bkp ".dump" | sqlite3 /nix/var/nix/db/db.sqlite
```

> ### How do I fix: error: current Nix store schema is version 10, but I only support 7

Reviewer: ```suggestion
### How to fix: `error: current Nix store schema is version 10, but I only support 7`
```

> This means you have upgraded Nix sqlite schema to a newer version, but then tried
> to use older Nix.

Reviewer: ```suggestion
This is a [known issue](https://github.com/NixOS/nix/issues/1251).

It means that using a new version of Nix upgraded the SQLite schema of the [database](https://nixos.org/manual/nix/stable/glossary#gloss-nix-database), and then you tried to use an older version Nix.
```

Use project names as they would call themselves.

> The solution is to dump the db and use old Nix version to initialize it:

Reviewer: ```suggestion
The solution is to dump the database, use the old Nix version to initialize it, and then re-import the data:
```

```shell-session
$ /path/to/nix/unstable/bin/nix-store --dump-db > /tmp/db.dump
$ mv /nix/var/nix/db /nix/var/nix/db.toonew
$ mkdir /nix/var/nix/db
$ nix-store --init # this is the old nix-store
$ nix-store --load-db < /tmp/db.dump
```

> ### I'm getting: writing to file: Connection reset by peer

Reviewer: ```suggestion
### How to fix: `writing to file: Connection reset by peer`
```

> 
> Too big files in src, out of resources (HDD space, memory)

Reviewer: This is missing a suggestion on what to do.

```suggestion

This may mean you are trying to import a too large file or directory into the [Nix store](https://nixos.org/manual/nix/stable/glossary#gloss-store), or your machine is running out of resources, such as disk space or memory.

Try to reduce the size of the directory to import, or run [garbage collection](https://nixos.org/manual/nix/stable/command-ref/nix-collect-garbage).
```
